embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/LogOrchestrator.kt

```diff
@@ -0,0 +1,74 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.comms.delivery.DeliveryService
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.worker.ScheduledWorker
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class LogOrchestrator(
+    private val logOrchestratorScheduledWorker: ScheduledWorker,
+    private val clock: Clock,
+    private val sink: LogSink,
+    private val deliveryService: DeliveryService
+) {
+    private var lastLogTime: Long = 0
+    private var firstLogInBatchTime: Long = 0
+    private var scheduledCheckFuture: ScheduledFuture<*>? = null
+
+    init {
+        sink.callOnLogsStored(::onLogsAdded)
+    }
+    private fun onLogsAdded() {
+        lastLogTime = clock.now()
+        if (firstLogInBatchTime == 0L) {
+            firstLogInBatchTime = lastLogTime
+        }
+        if (logsShouldBeSent()) {
+            sendLogs()
+        } else {
+            scheduleCheck()
+        }
+    }
+
+    private fun logsShouldBeSent(): Boolean {
+        return (
+            (sink.completedLogs().size >= MAX_LOGS_PER_BATCH) ||
+                (clock.now() - lastLogTime > MAX_INACTIVITY_TIME) ||
```

Comment:
Calling `now()` twice could result in different values if we are JUST at the end of the next millisecond. What about calling it once and using the result in both clauses so it's more consistent?

Reply:
Fixed